### PR TITLE
bluespec unstable-2020.02.09 -> unstable-2020.11.04

### PR DESCRIPTION
**Pull request by jcumming**
#  Continue on with #84142 
  - (thanks @flokli).
  - a small patch to install the STP stub (upstream PR: https://github.com/B-Lang-org/bsc/pull/278)
  - set LD_LIBRARY_PATH during build so that bsc finishes bootstrapping
  - use nixpkgs yices.src instead of a subrepo
  - match yices gmp-static idiom

I ran this through https://github.com/jcumming/bsc-testsuite, and the 13 failures appear to mostly be nixos-isms.

I'm waiting on to see if upstream takes the patch, so I can drop it from this PR. 

###### Motivation for this change

Update bluespec to latest version.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
- [x] ran bsc-testsuite

### CODE REVIEW

File: pkgs/development/compilers/bluespec/default.nix [hunk 1]
```diff
@@ -54,13 +61,13 @@ in stdenv.mkDerivation rec {
...
-    # drop stp support https://github.com/B-Lang-org/bsc/pull/31
-    (fetchpatch {
-      url = "https://github.com/flokli/bsc/commit/0bd48ecc2561541dc1368918863c0b2f4915006f.patch";
-      sha256 = "0bam9anld33zfi9d4gs502g94w49zhl5iqmbs2d1p5i19aqpy38l";
-    })
-  ];
+
+  postUnpack = ''
+    mkdir -p $sourceRoot/src/vendor/yices/v2.6/yices2
+    # XXX: only works because yices.src isn't a tarball.
```
Comment (jcumming) [inline pkgs/development/compilers/bluespec/default.nix:67]: FWIW; is there a better way to do this?

Comment (flokli) [reply pkgs/development/compilers/bluespec/default.nix:67]: Does the source code need to be there, or can we patch the build system to accept an external yices, and set paths to header files and libraries (or have the buildsystem ask pkgconfig for it)

Comment (jcumming) [reply pkgs/development/compilers/bluespec/default.nix:67]: Good point. I'll try that.

**Review by thoughtpolice — CHANGES_REQUESTED** [review]
Thanks! If you can just please fix the trailing whitespace/editorconfig issues, I think we can go ahead and merge this and keep clean up things in future PRs. Otherwise this looks fine to me, and it's good to keep bluespec working for everyone...

**Comment by thoughtpolice** [comment]
@ofborg build bluespec

**Comment by thoughtpolice** [comment]
@jcumming I've pushed this commit, along with a fix for an incorrect `sha256` hash, as part of d402d913d754469f421572284e00292f83de1af5 (with credit to you, of course.) I suspect the hash thing happened because source attributes are fixed-output, so if you download the source once you can screw up the hash later and it'll still succeed, unfortunately.

I've tested this package on my Ryzen machine however and it seems to build just fine. Let's keep submitting any improvements incrementally, either to upstream or here!

**Pull request closed without merging**